akka-typed/src/test/scala/akka/typed/ActorContextSpec.scala

```diff
@@ -426,6 +437,29 @@ class ActorContextSpec extends TypedSpec(ConfigFactory.parseString(
       }
     })
 
+    def `13 must terminate upon not handling Terminated`(): Unit = sync(setup("ctx13") { (ctx, startWith) ⇒
+      val self = ctx.self
+      startWith.mkChild(None, ctx.spawnAdapter(ChildEvent), self).keep {
+        case (subj, child) ⇒
+          subj ! Watch(child, self)
+      }.expectMessageKeep(500.millis) {
+        case (msg, (subj, child)) ⇒
+          msg should ===(Watched)
+          subj ! BecomeCareless(self)
+      }.expectMessageKeep(500.millis) {
+        case (msg, (subj, child)) ⇒
+          msg should ===(BecameCareless)
+          child ! Stop
+      }.expectFailureKeep(500.millis) {
+        case (f, (subj, child)) ⇒
+          f.child should ===(subj)
+          Failed.Stop
+      }.expectMessage(500.millis) {
+        case (msg, (subj, child)) ⇒
+          msg should ===(GotSignal(PostStop))
+      }
+    })
```

Comment:
ah, it's been a while since we had a commit in typed, reminds me how nice to read these are :)